src/jit/decomposelongs.cpp

```diff
@@ -920,64 +924,319 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftByOp = tree->gtGetOp2();
+
     genTreeOps oper = tree->OperGet();
+    genTreeOps shiftByOper = oldShiftByOp->OperGet();
 
     assert((oper == GT_LSH) || (oper == GT_RSH) || (oper == GT_RSZ));
 
     LIR::Use loOp1Use(Range(), &gtLong->gtOp.gtOp1, gtLong);
-    loOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
+    unsigned loOp1LclNum = loOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
 
     LIR::Use hiOp1Use(Range(), &gtLong->gtOp.gtOp2, gtLong);
-    hiOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
+    unsigned hiOp1LclNum = hiOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
 
-    LIR::Use shiftWidthUse(Range(), &tree->gtOp.gtOp2, tree);
-    shiftWidthUse.ReplaceWithLclVar(m_compiler, m_blockWeight);
+    LIR::Use shiftByUse(Range(), &tree->gtOp.gtOp2, tree);
+    shiftByUse.ReplaceWithLclVar(m_compiler, m_blockWeight);
 
     GenTree* loOp1 = gtLong->gtGetOp1();
     GenTree* hiOp1 = gtLong->gtGetOp2();
 
-    GenTree* shiftWidthOp = tree->gtGetOp2();
+    GenTree* shiftByOp = tree->gtGetOp2();
 
     Range().Remove(gtLong);
     Range().Remove(loOp1);
     Range().Remove(hiOp1);
 
-    Range().Remove(shiftWidthOp);
+    Range().Remove(shiftByOp);
 
-    // TODO-X86-CQ: If the shift operand is a GT_CNS_INT, we should pipe the instructions through to codegen
-    // and generate the shift instructions ourselves there, rather than replacing it with a helper call.
+    // If we are shifting by a constant int, we do not want to save to a helper, instead, we decompose.
```

Comment:
>save [](start = 63, length = 4)

(Minor) should comment read as ".. do not want to use a helper" instead of ".. do not want to save to a helper"?